tests/commands/test_store_charmlibs.py

```diff
@@ -0,0 +1,331 @@
+                {"metadata_patch": "", "metadata_api": ""},
+                r"^Library '.+' is missing the mandatory metadata fields: LIBAPI, LIBPATCH\.$",
+                id="missing api and patch",
+            ),
+        ],
+    )
+    @pytest.mark.skipif(sys.platform == "win32", reason="Windows not [yet] supported")
+    def test_invalid_metadata_field(self, charm_path, fields, match):
+        """Some metadata field is not really valid."""
+        test_path = self.create_lib(**fields)
+        with pytest.raises(CraftError, match=match):
+            get_lib_info(lib_path=test_path)
+
+    bad_field_match = r"^Library .+ metadata field {} is not zero or a positive integer\.$"
+
+    @pytest.mark.parametrize(
+        "lib_fields,match",
+        [
+            pytest.param(
+                {"metadata_api": "LIBAPI = v3"},
+                bad_field_match.format("LIBAPI"),
+                id="API non-integer",
+            ),
+            pytest.param(
+                {"metadata_api": "LIBAPI = -3"},
+                bad_field_match.format("LIBAPI"),
+                id="API negative",
+            ),
+            pytest.param(
+                {"metadata_patch": "LIBPATCH = beta3"},
+                bad_field_match.format("LIBPATCH"),
+                id="PATCH non-integer",
+            ),
+            pytest.param(
+                {"metadata_patch": "LIBPATCH = -1"},
+                bad_field_match.format("LIBPATCH"),
+                id="PATCH negative",
+            ),
+            pytest.param(
+                {"metadata_api": "LIBAPI = 0", "metadata_patch": "LIBPATCH = 0"},
+                r"^Library .+ metadata fields LIBAPI and LIBPATCH cannot both be zero\.$",
+                id="API and PATCH both zero",
+            ),
+        ],
+    )
+    def test_api_patch_invalid(self, lib_fields, match):
+        """Invalid values for API or patch versions."""
+        test_path = self.create_lib(**lib_fields)
+        with pytest.raises(CraftError, match=match):
+            get_lib_info(lib_path=test_path)
+
+    def test_metadata_api_different_path_api(
+        self,
+    ):
+        """The API value included in the file is different from the one in the path."""
+        test_path = self.create_lib(metadata_api="LIBAPI = 99")
+        match = r"^Library .+ metadata field LIBAPI is different from the version in the path\.$"
+        with pytest.raises(CraftError, match=match):
+            get_lib_info(lib_path=test_path)
+
+    @pytest.mark.parametrize(
+        "metadata_id",
+        [
+            pytest.param("LIBID = 99", id="Non-string"),
+            pytest.param("LIBID = 'moño'", id="Non-ASCII"),
+            pytest.param("LIBID = ''", id="empty"),
+        ],
+    )
+    def test_libid_invalid(self, metadata_id):
+        """The ID is not really a string."""
+        test_path = self.create_lib(metadata_id=metadata_id)
+        match = r"Library .+ metadata field LIBID must be a non-empty ASCII string\."
```

Comment:
Here and in other tests, do not build generic strings when checking for errors or other messages exposed to the user.

For example in this case, if we introduce a bug in the code and the error message is `Library foobarBROKEN metadata field LIBID must be a non-empty ASCII string.` this still will pass, and it shouldn't.

Yes, the chances are low, but we have to be extra careful with messages shown to the final user.